gen/cycle.ml

```diff
@@ -708,7 +715,7 @@ let set_same_loc st n0 =
 
   let tr_value e v = E.tr_value e.atom v
 
-  let rec do_set_write_val st pte_val = function
+  let rec do_set_write_val ?(rv = -1) st pte_val = function
```

Comment:
I do not understand the logic behind this optional argument `rv`. The state argument `st` has two components of interest, value `v` (in practice `v` is the value bound to `Ord`  in `st.map`) and component `co_cell`. Component `co_cell` (here, in the non-neon case `co_cell.(0)`) represents the contents of memory after the write is performed. As regards our extension component v still follows the sequence if integers, 0,1,2,... as before. The computation of `co_cell` changes: it becomes `set.co_cell.(0)` op `v` (op is the operation performed by the AMO operation and `v` is the operand of the instructio, available from `set_cell` as `n.evt.v`,

Hence, the function `set_cell` does not need the extra `rv` argument as the old value of memory is available as `st.cell.(0)` and the operand is available as `n.evt.v`.

Notice that further adaptation may be  needed for mixed-size mode: the calls to functions `tr_value` at lign 741/748 below and `E.overwrite_value` in `set_cell` are concern mixed-size mod and are trivial in non-mixed-size mode. But mixed-size is yet another story.